### DISCUSSION TITLE
Installation

### DISCUSSION THREAD

**Discussion by geri167** [GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200#12] [General]
So let me ask: 
the froeling.yaml file i start in the windows cmd "eingabeaufforderung" with:
esphome run froeling.yaml           this is all?

and then->

when i then only need the parameters from "kessel.jaml' i copy this in my /homeassistant/configuration.yaml??? 
for get the Entitäten from Kessel?

Sorry for the stupid questions

**Comment by GyroGearl00se** [comment]
The froeling.yaml contains the following section:

packages:
  remote_package:
    url: https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200
    ref: main
    files: [austragung.yaml, boiler_01.yaml, heizkoerper.yaml, kessel.yaml, puffer_01.yaml, zirkulationspumpe.yaml]
    refresh: 1d

This will take care of getting the other yaml files and include them.

The separation of the yaml files was not necessary - I just decided to do it this way to have it a little mit structured / organized.


Alternatively you could also download the yaml which you need/want and configure it to use them from your local storage.
You can find an example here:
https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/discussions/10



Now looking at it - I might set refresh to "never" to avoid your code being update automatically. :)

**Reply by geri167** [reply]
ok but in my case this comes

 
![image](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/137945931/6cdfba57-8373-44a7-9ff3-459e8ea652f4)

**Reply by GyroGearl00se** [reply]
Seems like you either have to install git on your computer or you bypass this with the alternative solution as described here:

https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/discussions/10#discussioncomment-8121238


Even easier would it be if you have Home Assistant and install ESPhome as a add-on.

**Reply by geri167** [reply]
OK, i do it with python and cmd and now also with ESPhome add on, always the same thing. 
![image](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/137945931/3fa642c4-dd27-4c15-b8dc-fae4e90d20f2)
no data,
![image](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/137945931/b3429392-b265-4713-9dbf-f8128a518800)
![image](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/137945931/5bf61a3f-24f8-4671-85bd-9449b2e3da52)
ESPHOME 
![image](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/137945931/e4f1bbc6-6918-417c-8c00-558836ac1e26)
should i take adresse: 1  on the fröling boiler and also 
modbus_controller:
  - id: froeling
    address: 2    ????
![image](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/137945931/bad11e03-bc61-464f-b1f5-f2ae1ce82a54)


![image](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/137945931/65b7dd74-eed1-444f-b703-e821cfc14ff6)

![image](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/137945931/6ddf2d8e-068d-4f66-bd28-3390a070fd42)

**Comment by GyroGearl00se** [comment]
Looks like the inclusion of the yam files from "local" is working.

Now the question is - why it won't read any data from your system.

Did you enable in the settings of your heating unit the Modbus RTU for COM2 & Modbus Protokoll 2014 ?
In the settings menu you should also be able to see the slave address. In my case "2" was set by default. I haven't changed it. Just make sure it is the same in code and on the device.

Which UART/RS232 adapter are you using ? Can you provide a picture of how you've wired it ?
A common issue with this is also the connection of TX & RX. Sometimes it helps to swap them around.

If all this is not going to help - Then it could be also the cable you're using between the adapter and the COM2.
There are 2 different types of cables. Straight-Trough or Crossover (Nullmodem).

For my understanding the cable we need in our case is the "Nullmodem". (See ModBus Lambdatronic 3200...pdf in this repository - 1.1 Datenkabel)

**Comment by wavoigt** [comment]
Normally, a cable with a female and a male plug is a straight-through cable, and a cable with 2 female plugs is a Nullmodem (or crossover) cable

**Comment by GyroGearl00se** [comment]
Alternatively you could also plug in the adapter directly into the COM port without using an additional cable,
I personally prefer using a cable to ensure that I can mount the ESP32 outside of the housing for better WiFi signal strength.

**Reply by geri167** [reply]
hi thanks for your feedback: 
to your questions:  Boiler settings,yes 
![20240122_074831](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/137945931/0f52e9e1-f093-432d-bb17-e6414dce8151)


i use the adapter an cable under discussion 11:
[](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/discussions/11 )

swap tx and rx cable between the d1 and RS232 adapter ????

Ok i use a normal sub without nullmodem. 

*do the light of L and or R1 flash if connection?? 

here your pictures
![20240122_074302](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/137945931/a15db386-f866-4c4e-8bb6-529575c5af03)

![20240122_074250](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/137945931/16d09ffb-cf1e-4d1d-b05a-38950df9d718)

Thanks for your help!

**Reply by GyroGearl00se** [reply]
I have a slightly different rs232 adapter than yours. On mine - there isn't blinking anything at all. Just works.

I updated just now the wiring diagramm and the froeling.yaml (using UART2 instead of UART0 => different GPIO)
Also updated the wiring diagram accordingly.

Settings on your boiler look fine.

At the moment I can only assume that the cable you're using might be the wrong one.

**Reply by GyroGearl00se** [reply]
... or the adapter itself is faulty.

swapping TX & RX between ESP32 & the RS232 adapter is always worth a try :) Once swapped - give it ~ 1minute time to refresh as the update_intervall is configured to 60s.

**Reply by geri167** [reply]
so i swapped, no chance!

so i installed 10times all new, no chance

so i done the UART2, and adapt fröling.yaml, no chance, but something is new -> i cannot reach froeling.lokal:
![image](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/137945931/29368e72-b42e-4a33-a70d-53f91ea1948f)

here my log, maybe you can see something:
[logs_froeling_logs (2).txt](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/files/14030531/logs_froeling_logs.2.txt)

for now i have no idea, 

Maybe it is the rs232 adapter, but yours is not be disponible.

**Reply by geri167** [reply]
here my yaml: 
[froeling.txt](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/files/14030586/froeling.txt)

**Reply by GyroGearl00se** [reply]
With your IP I take it that you have a Fritz Box. I faced similar issue once with reaching devices within my network. You could also try "froeling.fritz.box".
Anyway... This is not related to your issue.

At the moment I can still only assume that either the cable is wrong and/or the rs232 adapter is faulty.

What I also just noticed now is from your previously sent picture:

![image](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/94406798/19345ad9-52b9-4f90-8fe4-582cc2e9ff31)

Not sure if this is relevant... But I found this kind of boards on amazon with the following article: https://www.amazon.de/ANGEEK-Serial-Converter-Connector-MAX3232/dp/B07ZDK4BLH/ -> See 2nd picture. Looks same as yours.

Compare to this one here:
https://www.amazon.de/ARCELI-Root-Modul-Anschluss-Chip-TTL-Buchse-DB9-Konverter-Modulplatine-Ausr%C3%BCstungs-Upgrades/dp/B09DYDFZRW/

This one has the C5 capacitor which is missing on yours.
I can't say for sure if this is the issue. Just saying it as I noticed it.

**Reply by geri167** [reply]
a sorry i forget, i added a "nullmodem" Adapter so this is not the issue!

**Reply by geri167** [reply]
so i made an update, so now the ip is good, 
and there i see this 
![image](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/137945931/6812e25b-7a18-4829-a218-510f2ac1eb95)

I also see homeassistant start and go down very often, since i done this....

![image](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/137945931/be3cd03d-1c41-4796-b631-8569d8af09b9)

![image](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/137945931/3c28ecfc-aabe-4682-9343-cc551289df91)

**Comment by wavoigt** [comment]
I have the ANGEEK boards. I replaced the MAX3232 because i was in doubt with the Fake parts.
Mine is working fine. It may also work with the fake chip, but no guarantee.
See also the reviews on Amazon: https://www.amazon.de/product-reviews/B07ZDK4BLH/ref=acr_dp_hist_1?ie=UTF8&filterByStar=one_star&reviewerType=all_reviews#reviews-filter-bar
You can get the originals here: https://www.reichelt.de/rs232-2-treiber-2-empfaenger-so-16-max-3232-cse-p57869.html?&trstct=pos_0&nbc=1
![MAX3232 Fake - Origin](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/24320737/b4fda46d-5ea8-4cc6-8cc2-506963fba6fb)

**Comment by geri167** [comment]
what was your issue ?Von meinem/meiner Galaxy gesendet
-------- Ursprüngliche Nachricht --------Von: Walter ***@***.***> Datum: 24.01.24  10:15  (GMT+01:00) An: GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200 ***@***.***> Cc: geri167 ***@***.***>, Author ***@***.***> Betreff: Re: [GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200] Installation (Discussion #12) 
I have the ANGEEK boards. I replaced the MAX3232 because i was in doubt with the Fake parts.
Mine is working fine.
MAX3232.Fake.-.Origin.jpg (view on web)

**Reply by wavoigt** [reply]
I also got no data, but there was another reason, i had a ESP8266 board instead ESP32.

**Comment by Andreas-84** [comment]
Hi,
i have the same issue like geri167 with strange values:
![image](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/156196803/d53a9663-773c-4ef7-a3f8-2ee943ef6cc8)

Everything is set like in description.

My question is, if a new adapter has solved the problem? @geri167 

Thanks!

I have this board:
https://www.amazon.de/gp/product/B071P98VTG/ref=ppx_yo_dt_b_search_asin_image?ie=UTF8&psc=1

This cable
https://www.amazon.de/DTECH-RS232-Serienkabel-Stecker-Buchse-Nullmodemkabel-7-Draht-Frequenzweiche-Datenkommunikation/dp/B08YNTN9H7

This adapter:
https://www.ebay.de/itm/354585556443?hash=item528ef235db:g:NxAAAOSwSd1kHHHx

**Reply by geri167** [reply]
hi @Andreas-84 yes the new adapter solved the problem

**Comment by wavoigt** [comment]
Do NOT use a Nullmodem cable.
a normal 1:1 cable will work.

**Reply by Andreas-84** [reply]
Thanks I will try that!
